src/charactersheet/viewmodels/character/items/container.js

```diff
@@ -0,0 +1,131 @@
+        this.existingData = params.data ? params.data : null;
+        this.containerId = ko.utils.unwrapObservable(params.containerId);
+        this.collapseAllId = `.${this.containerId}`;
+        this.entity = ko.observable();
+        this.modelClass.bind(this);
+        autoBind(this);
+    }
+
+    modelClass() {
+        return Item;
+    }
+
+    generateBlank() {
+        const thisClazz = this.modelClass();
+        const newEntity = new thisClazz();
+        const coreKey = CoreManager.activeCore().uuid();
+        newEntity.coreUuid(coreKey);
+        return newEntity;
+    }
+
+    emptyContainer() {
+        this.entities().map(this.removeFromContainer);
+    }
+
+    async removeFromContainer(item) {
+        item.parent(null);
+        await item.save();
+        this.removeFromList(item);
+    }
+
+    async load() {
+        this.entity(this.generateBlank());
+        await this.refresh();
+        await super.load();
+        if (this.containerId) {
+            $(`#${this.containerId}`).on('hidden.bs.collapse', this.collapseAll);
+    
+        }
+    }
+
+    async refresh() {
+        if (this.existingData) {
+            this.entity().importValues(this.existingData.exportValues());
+        } else {
+            const coreKey = CoreManager.activeCore().uuid();
+            await this.entity().load({uuid: coreKey});
+        }
+        this.entities = this.entity().children;
+    }
+
+    sorts() {
+        return {
+            ...super.sorts(),
+            'quantity asc': { field: 'quantity', direction: 'asc', numeric: true},
+            'quantity desc': { field: 'quantity', direction: 'desc', numeric: true},
+            'weight asc': { field: 'weight', direction: 'asc', numeric: true},
+            'weight desc': { field: 'weight', direction: 'desc', numeric: true},
+            'totalCalculatedWeight asc': { field: 'totalCalculatedWeight', direction: 'asc', numeric: true},
+            'totalCalculatedWeight desc': { field: 'totalCalculatedWeight', direction: 'desc', numeric: true},
+            'cost asc': { field: 'cost', direction: 'asc', numeric: true},
+            'cost desc': { field: 'cost', direction: 'desc', numeric: true},
+            'totalCalculatedCost asc': { field: 'totalCalculatedCost', direction: 'asc', numeric: true},
+            'totalCalculatedCost desc': { field: 'totalCalculatedCost', direction: 'desc', numeric: true}
+        };
+    }
+
+    collapseAll() {
+        $(this.collapseAllId + ' .collapse.in').collapse('hide');
+    }
+
+    contains(item) {
+        return this.entities().find(e=>e.uuid() === item.uuid()) !== undefined;
+    }
+
+    isTheParentOf(item) {
+        return !!item.hasParent() && this.entity().url() === item.parent();
+    }
+
+    addToList(item) {
+        if (item && this.isTheParentOf(item)) {
+            super.addToList(item);
+            Notifications.item.changed.dispatch(this.entity());
+        }
+    }
+
+    replaceInList(item) {
+        if (item) {
```

Comment:
For readability, can we use early returns? If possible, do this for all the `ifs`
```
if (!item) {
	return;
}
```

Reply:
Fixed